### PR TITLE
Use ngx-translate

### PR DESCRIPTION
**Pull request by Senyoret1**
With this pr the explorer now has most of the texts centralized in the ./assets/i18n/en.json file. With this change it is now much easier to make changes to the texts that are used in many parts of the explorer, as well as making customizations for other coins. Additionally, it is possible to create translations by simply adding and translating files such as "fr.json" (French) and "cn.json" (Chinese), but no ui controls for selecting the languaje have been added in this pr.

To avoid maintainability problems, the text on the header and the footer were not included, since those controls are shared with the website. The appearance of the explorer remains the same.

Taking all this into account, this pr partially solves https://github.com/skycoin/skycoin-explorer/issues/165 . There are still more things that could be done, but I think they should be requested creating a new issue.

### CODE REVIEW

**Comment by gz-c** [comment]
If a fiber coin wanted to reuse our explorer, they would need to make separate changes for the header and footer?

File: src/app/components/pages/address-detail/address-detail.component.html [hunk 1]
```diff
@@ -30,45 +30,45 @@ <h2 class="-xs-only">Address</h2>
     <div class="row">
       <div class="col-md-10 col-sm-12">
         <div class="-row">
-          <div class="-float-left">Transaction ID<span class="-xs-sm-only">:</span></div><br class="-xs-sm-only"/>
+          <div class="-float-left">{{ 'txBoxes.transsactionID' | translate }}<span class="-xs-sm-only">:</span></div><br class="-xs-sm-only"/>
```
Comment (gz-c) [inline src/app/components/pages/address-detail/address-detail.component.html]: typo `transsactionID` -> `transactionID`

File: src/app/components/pages/address-detail/address-detail.component.html [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
       <app-qr-code [string]="address" *ngIf="address"></app-qr-code>
-      <span>Address</span><br class="-xs-only" /> {{ address ? address : loadingMsg }}
+      <span>{{ 'general.address' | translate }}</span><br class="-xs-only" /> {{ address ? address : loadingMsg }}
     </div>
-    <div class="-row -tx-number"><span># of Transactions</span><br class="-xs-only" /><div> {{ transactions ? transactions.length : loadingMsg }} </div></div>
-    <div class="-row"><span>Total Received</span><br class="-xs-only" /><div> {{ transactions ? ( totalReceived | number:'1.0-6') : loadingMsg }} </div></div>
-    <div class="-row"><span>Current Balance</span><br class="-xs-only" /><div> {{ transactions ? ( balance | number:'1.0-6') : loadingMsg }} </div></div>
-    <div class="-row"><span>Tools</span><br class="-xs-only" /><div> <a [routerLink]="'/app/unspent/' + address" class="-link" *ngIf="address">Unspent outputs</a> <span *ngIf="!address">{{ loadingMsg }}</span> </div></div>
+    <div class="-row -tx-number"><span>{{ 'addressDetail.TxsNumber' | translate }}</span><br class="-xs-only" /><div> {{ transactions ? transactions.length : loadingMsg }} </div></div>
```
Comment (gz-c) [inline src/app/components/pages/address-detail/address-detail.component.html]: `TxsNumber` should not be capitalized?

File: src/app/components/pages/richlist/richlist.component.html [hunk 3]
```diff
@@ -1,17 +1,17 @@
-<h2>Rich List</h2>
+<h2>{{ 'ritchlist.title' | translate }}</h2>
```
Comment (gz-c) [inline src/app/components/pages/richlist/richlist.component.html]: `richlist`

**Comment by Senyoret1** [comment]
I did some testing and the cache did not cause problems. However, just in case, I anyway implemented the recommended method for cache busting located at the bottom of the readme of this repository: https://github.com/ngx-translate/http-loader . I did some more testing after that and it worked fine.

**Pull request merged**